File: m2cgen/interpreters/java/linear_algebra.java
public static double[] addVectors(double[] v1, double[] v2) {
    double[] result = new double[v1.length];

    for (int i = 0; i < v1.length; i++) {
        result[i] = v1[i] + v2[i];
    }

    return result;
}

public static double[] mulVectorNumber(double[] v1, double num) {
    double[] result = new double[v1.length];

    for (int i = 0; i < v1.length; i++) {
        result[i] = v1[i] * num;
    }

    return result;
}

File: m2cgen/interpreters/lua/linear_algebra.lua
function add_vectors(v1, v2)
    local result = {}
    for i = 1, #v1 do
        result[i] = v1[i] + v2[i]
    end
    return result
end
function mul_vector_number(v1, num)
    local result = {}
    for i = 1, #v1 do
        result[i] = v1[i] * num
    end
    return result
end